main.tf

```diff
@@ -348,8 +348,8 @@ resource "kubernetes_deployment" "main" {
             }
 
             resources {
-              limits   = var.deployment.resource_limits
-              requests = var.deployment.resource_requests
+              limits   = init_container.value["resource_limits"] != null ? init_container.value["resource_limits"] : var.deployment.resource_limits
```

Comment:
We should not use the variable passed in the variable, for example if we don't want to specify it, so let's default to null otherwise use the passed value